src/zenml/materializers/numpy_materializer.py

```diff
@@ -11,236 +11,27 @@
-        self, arr: "NDArray[Any]"
-    ) -> Dict[str, "MetadataType"]:
-        """Extract metadata from the given numpy array.
-
-        Args:
-            arr: The numpy array to extract metadata from.
-
-        Returns:
-            The extracted metadata as a dictionary.
-        """
-        if np.issubdtype(arr.dtype, np.number):
-            return self._extract_numeric_metadata(arr)
-        elif np.issubdtype(arr.dtype, np.unicode_) or np.issubdtype(
-            arr.dtype, np.object_
-        ):
-            return self._extract_text_metadata(arr)
-        else:
-            return {}
-
-    def _extract_numeric_metadata(
-        self, arr: "NDArray[Any]"
-    ) -> Dict[str, "MetadataType"]:
-        """Extracts numeric metadata from a numpy array.
-
-        Args:
-            arr: The numpy array to extract metadata from.
-
-        Returns:
-            A dictionary of metadata.
-        """
-        min_val = np.min(arr).item()
-        max_val = np.max(arr).item()
-
-        numpy_metadata: Dict[str, "MetadataType"] = {
-            "shape": tuple(arr.shape),
-            "dtype": DType(arr.dtype.type),
-            "mean": np.mean(arr).item(),
-            "std": np.std(arr).item(),
-            "min": min_val,
-            "max": max_val,
-        }
-        return numpy_metadata
-
-    def _extract_text_metadata(
-        self, arr: "NDArray[Any]"
-    ) -> Dict[str, "MetadataType"]:
-        """Extracts text metadata from a numpy array.
-
-        Args:
-            arr: The numpy array to extract metadata from.
-
-        Returns:
-            A dictionary of metadata.
-        """
-        text = " ".join(arr)
-        words = text.split()
-        word_counts = Counter(words)
-        unique_words = len(word_counts)
-        total_words = len(words)
-        most_common_word, most_common_count = word_counts.most_common(1)[0]
-
-        text_metadata: Dict[str, "MetadataType"] = {
-            "shape": tuple(arr.shape),
-            "dtype": DType(arr.dtype.type),
-            "unique_words": unique_words,
-            "total_words": total_words,
-            "most_common_word": most_common_word,
-            "most_common_count": most_common_count,
-        }
-        return text_metadata
+"""Implementation of ZenML's numpy materializer."""
+
+# With the code below, we keep a reference to the NumpyMaterializer in its
+# original spot. So, if someone has an artifact version that was created with
+# the NumpyMaterializer, (assuming that they have it installed), they will still
+# be able to use the artifact version. However, if this file is to be removed,
+# you have to write a DB migration script to change the materializer source
+# of the old artifact version entries.
+
+from zenml.exceptions import IntegrationError
+
+try:
+    from zenml.integrations.numpy.materializers.numpy_materializer import (  # noqa
+        NumpyMaterializer,
+    )
+except (ImportError, ModuleNotFoundError) as e:
+    raise IntegrationError(
```

Comment:
IMO this should be an `ImportError` as that's the reason for the failure, which is very different from an issue with one of our integrations

Reply:
Changed it!